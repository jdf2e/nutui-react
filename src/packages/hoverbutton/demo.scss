.demo {
  .nutui-hoverbutton-clickable {
    &:active::before {
      opacity: 0.1;
    }

    cursor: pointer;

    &::before {
      position: absolute;
      top: 50%;
      left: 50%;
      width: 100%;
      height: 100%;
      background-color: $color-mask;
      border: inherit;
      border-color: $color-mask;
      border-radius: inherit;
      transform: translate(-50%, -50%);
      opacity: 0;
      content: ' ';
    }
  }
}

[dir='rtl'] .nut-hoverbutton,
.nut-rtl .nut-hoverbutton {
  .nut-icon-ArrowRight {
    transform: rotate(180deg);
  }
}
