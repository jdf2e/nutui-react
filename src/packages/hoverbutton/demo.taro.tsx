import React, { useState } from 'react'
import Taro from '@tarojs/taro'
import { ScrollView, View, Text } from '@tarojs/components'
import { Cell } from '@nutui/nutui-react-taro'
import { useTranslate } from '@/sites/assets/locale/taro'
import Header from '@/sites/components/header'
import Demo1 from './demos/taro/demo1'
import Demo1Rn from './demos/taro/demo1-rn'
import Demo2 from './demos/taro/demo2'
import Demo2Rn from './demos/taro/demo2-rn'
import Demo3 from './demos/taro/demo3'
import Demo3Rn from './demos/taro/demo3-rn'
import Demo4 from './demos/taro/demo4'
import Demo4Rn from './demos/taro/demo4-rn'
// import Demo5 from './demos/taro/demo5'

const isNative = Taro.getEnv() === 'RN' || Taro.getEnv() === 'HARMONY'

const HoverDemo = () => {
  const [translated] = useTranslate({
    'zh-CN': {
      show: '展示',
      basic: '基础用法',
      multiButtons: '多个按钮',
      hasTabbar: '有底部导航栏的情况',
      customZIndex: '自定义层级',
      customSpacing: '自定义间距',
    },
    'zh-TW': {
      show: '展示',
      basic: '基礎用法',
      multiButtons: '多個按鈕',
      hasTabbar: '有底部導航欄的情況',
      customZIndex: '自定義層級',
      customSpacing: '自定義間距',
    },
    'en-US': {
      show: 'Show ',
      basic: 'Basic Usage',
      multiButtons: 'Multiple Buttons',
      hasTabbar: 'With Tabbar',
      customZIndex: 'Custom Z-Index',
      customSpacing: 'Custom Spacing',
    },
  })
  const [curDemo, setCurDemo] = useState('basic')

  return (
    <View>
      <Header />
      <ScrollView
        className={`demo ${Taro.getEnv() === 'WEB' ? 'web' : ''}`}
        style={isNative ? { minHeight: 420 } : {}}
      >
        <Text className="h2">{translated.basic}</Text>
        <Cell
          title={`${translated.show}${translated.basic}`}
          onClick={() => {
            setCurDemo('basic')
          }}
        />

        <Text className="h2">{translated.multiButtons}</Text>
        <Cell
          title={`${translated.show}${translated.multiButtons}`}
          onClick={() => {
            setCurDemo('multiButtons')
          }}
        />

        <Text className="h2">{translated.hasTabbar}</Text>
        <Cell
          title={`${translated.show}${translated.hasTabbar}`}
          onClick={() => {
            setCurDemo('hasTabbar')
          }}
        />

        <Text className="h2">{translated.customZIndex}</Text>
        <Cell
          title={`${translated.show}${translated.customZIndex}`}
          onClick={() => {
            setCurDemo('customZIndex')
          }}
        />

        {/* @TODO RN、鸿蒙端暂不支持 ConfigProvider */}
        {/* {!isNative && <View className="h2">{translated.customSpacing}</View>}
        {!isNative && (
          <Cell
            title={`${translated.show}${translated.customSpacing}`}
            onClick={() => {
              setCurDemo('customSpacing')
            }}
          />
        )} */}
      </ScrollView>

      {curDemo === 'basic' && (isNative ? <Demo1Rn /> : <Demo1 />)}
      {curDemo === 'multiButtons' && (isNative ? <Demo2Rn /> : <Demo2 />)}
      {curDemo === 'hasTabbar' && (isNative ? <Demo3Rn /> : <Demo3 />)}
      {curDemo === 'customZIndex' && (isNative ? <Demo4Rn /> : <Demo4 />)}
      {/* {curDemo === 'customSpacing' && <Demo5 />} */}
    </View>
  )
}

export default HoverDemo
