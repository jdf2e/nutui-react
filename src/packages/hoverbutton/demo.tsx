import React, { useState } from 'react'
import { Button } from '@nutui/nutui-react-taro'
import { useTranslate } from '@/sites/assets/locale/taro'
import Header from '@/sites/components/header'
import Demo1 from './demos/h5/demo1'
import Demo2 from './demos/h5/demo2'
import Demo3 from './demos/h5/demo3'
import Demo4 from './demos/h5/demo4'
// import Demo5 from './demos/h5/demo5'

const HoverDemo = () => {
  const [translated] = useTranslate({
    'zh-CN': {
      show: '展示',
      basic: '基础用法',
      multiButtons: '多个按钮',
      hasTabbar: '有底部导航栏的情况',
      customZIndex: '自定义层级',
      customSpacing: '自定义间距',
    },
    'zh-TW': {
      show: '展示',
      basic: '基礎用法',
      multiButtons: '多個按鈕',
      hasTabbar: '有底部導航欄的情況',
      customZIndex: '自定義層級',
      customSpacing: '自定義間距',
    },
    'en-US': {
      show: 'Show ',
      basic: 'Basic Usage',
      multiButtons: 'Multiple Buttons',
      hasTabbar: 'With Tabbar',
      customZIndex: 'Custom Z-Index',
      customSpacing: 'Custom Spacing',
    },
  })
  const [curDemo, setCurDemo] = useState('basic')

  return (
    <>
      <Header />
      <>
        <h2>{translated.basic}</h2>
        <Button
          block
          onClick={() => {
            setCurDemo('basic')
          }}
        >
          {`${translated.show}${translated.basic}`}
        </Button>
        {curDemo === 'basic' && <Demo1 />}

        <h2>{translated.multiButtons}</h2>
        <Button
          block
          onClick={() => {
            setCurDemo('multiButtons')
          }}
        >
          {`${translated.show}${translated.multiButtons}`}
        </Button>
        {curDemo === 'multiButtons' && <Demo2 />}

        <h2>{translated.hasTabbar}</h2>
        <Button
          block
          onClick={() => {
            setCurDemo('hasTabbar')
          }}
        >
          {`${translated.show}${translated.hasTabbar}`}
        </Button>
        {curDemo === 'hasTabbar' && <Demo3 />}

        <h2>{translated.customZIndex}</h2>
        <Button
          block
          onClick={() => {
            setCurDemo('customZIndex')
          }}
        >
          {`${translated.show}${translated.customZIndex}`}
        </Button>
        {curDemo === 'customZIndex' && <Demo4 />}

        {/* <h2>{translated.customSpacing}</h2> */}
        {/* <Demo5 /> */}
      </>
    </>
  )
}

export default HoverDemo
