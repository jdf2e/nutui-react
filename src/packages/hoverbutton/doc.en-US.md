# HoverButton

## Introduction

Several functional buttons, typically appearing at the bottom right corner of a page, are floating and fixed in position, not offset by page scrolling; Icons can be replaced according to actual functionality.

## Installation

```tsx
import { HoverButton } from '@nutui/nutui-react-taro'
```

## Code Demonstration

### Basic Usage

:::demo

<CodeBlock src='taro/demo1.tsx'></CodeBlock>

:::

### Multiple Buttons

:::demo

<CodeBlock src='taro/demo2.tsx'></CodeBlock>

:::

### With Bottom Navigation Bar

:::demo

<CodeBlock src='taro/demo3.tsx'></CodeBlock>

:::

### Custom Level

:::demo

<CodeBlock src='taro/demo4.tsx'></CodeBlock>

:::

## HoverButton

### Props

| Property | Description | Type | Default Value |
| --- | --- | --- | --- |
| zIndex | Set component page level | `number` | `10` |
| tabbarHeight | Bottom navigation bar height (excluding safe area height) | `number` | `-` |
| icon | Set button icon | `ReactNode` | `-` |
| onClick | Button click event trigger | `Function` | `-` |

## HoverButton.Item

### Props

| Property | Description | Type | Default Value |
| --- | --- | --- | --- |
| icon | Set button icon | `ReactNode` | `-` |
| onClick | Button click event trigger | `Function` | `-` |

## Theme Customization

### Style Variables

The component provides the following CSS variables that can be used for custom styling, please refer to [ConfigProvider component](#/zh-CN/component/configprovider) for usage.

| Name | Description | Default Value |
| --- | --- | --- |
| \--nutui-hoverbutton-spacing | Button vertical spacing | `16px` |
| \--nutui-hoverbutton-position-bottom | Button area distance from the bottom of the screen | `48px` |
| \--nutui-hoverbutton-position-right | Button area distance from the right side of the screen | `16px` |
| \--nutui-hoverbutton-item-border-color | Button border color | `rgba(0, 0, 0, 0.06)` |
| \--nutui-hoverbutton-item-background | Button normal background color | `#FFFFFF` |
| \--nutui-hoverbutton-item-background-active | Button active background color | `#F6F6F6` |
| \--nutui-hoverbutton-item-icon-color | Icon normal color | `#1A1A1A` |
| \--nutui-hoverbutton-item-icon-color-active | Icon active color | `#595959` |