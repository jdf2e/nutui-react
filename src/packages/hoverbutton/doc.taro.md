# HoverButton

## 介绍

若干功能按钮，通常出现在页面右下角，悬浮且位置固定，不随页面滚动出现位置的偏移；
可根据实际功能替换图标。

## 安装

```tsx
import { HoverButton } from '@nutui/nutui-react-taro'
```

## 代码演示

### 基础用法

:::demo

<CodeBlock src='taro/demo1.tsx'></CodeBlock>

:::

### 多个按钮

:::demo

<CodeBlock src='taro/demo2.tsx'></CodeBlock>

:::

### 有底部导航栏的情况

:::demo

<CodeBlock src='taro/demo3.tsx'></CodeBlock>

:::

### 自定义层级

:::demo

<CodeBlock src='taro/demo4.tsx'></CodeBlock>

:::

## HoverButton

### Props

| 属性   | 描述             | 类型   | 默认值 |
| ------ | ---------------- | ------ | ------ |
| zIndex | 设置组件页面层级 | `number` | `10`     |
| tabbarHeight | 底部导航栏高度（不包含安全区高度）| `number` | `-` |
| icon    | 设置按钮图标       | ReactNode | `-`      |
| onClick | 按钮点击时触发事件 | Function  | `-`       |

## HoverButton.Item

### Props

| 属性    | 描述               | 类型      | 默认值 |
| ------- | ------------------ | --------- | ------ |
| icon    | 设置按钮图标       | `ReactNode` | `-`      |
| onClick | 按钮点击时触发事件 | `Function`  | `-`       |

## 主题定制

### 样式变量

组件提供了下列 CSS 变量，可用于自定义样式，使用方法请参考 [ConfigProvider 组件](#/zh-CN/component/configprovider)。

| 名称                                        | 说明                     | 默认值                |
| ------------------------------------------- | ------------------------ | --------------------- |
| \--nutui-hoverbutton-spacing                | 按钮垂直间距             | `16px`                |
| \--nutui-hoverbutton-position-bottom        | 按钮区域距离屏幕底部距离 | `48px`                |
| \--nutui-hoverbutton-position-right         | 按钮区域距离屏幕右侧距离 | `16px`                |
| \--nutui-hoverbutton-item-border-color      | 按钮边框色               | `rgba(0, 0, 0, 0.06)` |
| \--nutui-hoverbutton-item-background        | 按钮背景色-正常态        | `#FFFFFF`             |
| \--nutui-hoverbutton-item-background-active | 按钮背景色-点击态        | `#F6F6F6`             |
| \--nutui-hoverbutton-item-icon-color        | 图标色-正常态            | `#1A1A1A`             |
| \--nutui-hoverbutton-item-icon-color-active | 图标色-点击态            | `#595959`             |

