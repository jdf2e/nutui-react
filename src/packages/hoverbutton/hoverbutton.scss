.nut-hoverbutton {
  &-container {
    position: fixed;
    right: $hoverbutton-position-right;
    bottom: $hoverbutton-position-bottom;
    z-index: 10;
  }

  &-container-rn {
    position: absolute;
  }

  display: flex;
  flex-direction: column;
  gap: $hoverbutton-spacing;
}

[dir='rtl'] .nut-hoverbutton-container,
.nut-hoverbutton-container-rtl {
  right: auto;
  left: $hoverbutton-position-right;
}
