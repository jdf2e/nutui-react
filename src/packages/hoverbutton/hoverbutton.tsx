import React, { FunctionComponent } from 'react'
import classNames from 'classnames'
import { BasicComponent, ComponentDefaults } from '@/utils/typings'
import { useRtl } from '@/packages/configprovider/index'
import {
  HoverButtonItem,
  HoverButtonItemProps,
} from '../hoverbuttonitem/hoverbuttonitem'
import { SafeArea } from '../nutui.react'

export interface HoverButtonProps extends BasicComponent, HoverButtonItemProps {
  zIndex: number
  tabbarHeight?: number
}

const defaultProps = {
  ...ComponentDefaults,
} as HoverButtonProps

const classPrefix = 'nut-hoverbutton'

export const HoverButton: FunctionComponent<
  Partial<HoverButtonProps> &
    Omit<React.HTMLAttributes<HTMLDivElement>, 'title'>
> & {
  Item: typeof HoverButtonItem
} = (props) => {
  const { children, zIndex, tabbarHeight, className, style, icon, onClick } = {
    ...defaultProps,
    ...props,
  }

  const rtl = useRtl()

  const baseStyle = { ...style }

  if (tabbarHeight) {
    const bottom = tabbarHeight + 16
    baseStyle.bottom = `${bottom}px`
  }

  if (typeof zIndex !== 'undefined') {
    baseStyle.zIndex = zIndex
  }

  return (
    <div
      className={classNames([`${classPrefix}-container`, className], {
        [`${classPrefix}-container-rtl`]: rtl,
      })}
      style={baseStyle}
    >
      <div className={classPrefix}>
        {children ||
          (icon && <HoverButtonItem icon={icon} onClick={onClick} />)}
      </div>
      <SafeArea position="bottom" />
    </div>
  )
}

HoverButton.displayName = 'NutHoverButton'
HoverButton.Item = HoverButtonItem
