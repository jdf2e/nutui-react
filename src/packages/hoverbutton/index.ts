import { HoverButton } from './hoverbutton'

export type { HoverButtonProps } from './hoverbutton'
export default HoverButton
