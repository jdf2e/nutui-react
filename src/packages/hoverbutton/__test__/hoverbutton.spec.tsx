import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Cart } from '@nutui/icons-react'
import { HoverButton } from '../hoverbutton'

test('emit click event', () => {
  const handleClick = vi.fn()
  const { container } = render(
    <HoverButton data-testid="position" icon={<Cart />} onClick={handleClick} />
  )
  expect(container.querySelector('.nut-hoverbutton')?.childElementCount).toBe(1)
  fireEvent.click(container)
  expect(handleClick).toBeCalled
  expect(container).toMatchSnapshot()
})

test('child element count', () => {
  const { container } = render(
    <HoverButton>
      <HoverButton.Item icon={<Cart />} />
      <HoverButton.Item icon={<Cart />} />
      <HoverButton.Item icon={<Cart />} />
    </HoverButton>
  )
  expect(container.querySelector('.nut-hoverbutton')?.childElementCount).toBe(3)
  expect(container).toMatchSnapshot()
})

test('tabbar height', () => {
  const { container } = render(
    <HoverButton icon={<Cart />} tabbarHeight={48} />
  )

  expect(container.querySelector('.nut-hoverbutton-container')).toHaveAttribute(
    'style',
    'bottom: 64px;'
  )
  expect(container).toMatchSnapshot()
})
