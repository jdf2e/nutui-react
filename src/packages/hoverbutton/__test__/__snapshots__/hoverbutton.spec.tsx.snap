// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`child element count 1`] = `
<div>
  <div
    class="nut-hoverbutton-container"
  >
    <div
      class="nut-hoverbutton"
    >
      <div
        class="nut-hoverbutton-item-container"
      >
        <div
          class="nut-hoverbutton-item-icon"
        >
          <svg
            aria-labelledby="Cart"
            class="nut-icon nut-icon-Cart "
            role="presentation"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 65.17c0-24.21 19.37-43.84 43.24-43.84h51.14c70.46 0 128.85 55.4 133.55 126.68l.74 11.12h661.46c81.45 0 144 73.15 132.48 154.9l-33.02 234.45c-14.63 103.94-97.49 184.23-200.58 194.35l-467.47 45.86C246.4 796.07 180.12 739 175.1 662.63l-30.03-456.32v-.76l-3.42-51.69c-1.66-25.24-22.34-44.84-47.27-44.85H43.24C19.35 109.01 0 89.38 0 65.17m234.45 181.66 26.95 409.98c1.77 27.03 25.22 47.21 51.82 44.59l467.47-45.87c63.36-6.19 114.28-55.55 123.29-119.44L937 301.63c4.1-28.93-18.05-54.83-46.87-54.82h-655.7zm64.22 670.5a85.33 85.33 0 1 1-170.67 0 85.33 85.33 0 0 1 170.67 0M896 1002.67A85.33 85.33 0 1 0 896 832a85.33 85.33 0 0 0 0 170.67"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      <div
        class="nut-hoverbutton-item-container"
      >
        <div
          class="nut-hoverbutton-item-icon"
        >
          <svg
            aria-labelledby="Cart"
            class="nut-icon nut-icon-Cart "
            role="presentation"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 65.17c0-24.21 19.37-43.84 43.24-43.84h51.14c70.46 0 128.85 55.4 133.55 126.68l.74 11.12h661.46c81.45 0 144 73.15 132.48 154.9l-33.02 234.45c-14.63 103.94-97.49 184.23-200.58 194.35l-467.47 45.86C246.4 796.07 180.12 739 175.1 662.63l-30.03-456.32v-.76l-3.42-51.69c-1.66-25.24-22.34-44.84-47.27-44.85H43.24C19.35 109.01 0 89.38 0 65.17m234.45 181.66 26.95 409.98c1.77 27.03 25.22 47.21 51.82 44.59l467.47-45.87c63.36-6.19 114.28-55.55 123.29-119.44L937 301.63c4.1-28.93-18.05-54.83-46.87-54.82h-655.7zm64.22 670.5a85.33 85.33 0 1 1-170.67 0 85.33 85.33 0 0 1 170.67 0M896 1002.67A85.33 85.33 0 1 0 896 832a85.33 85.33 0 0 0 0 170.67"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      <div
        class="nut-hoverbutton-item-container"
      >
        <div
          class="nut-hoverbutton-item-icon"
        >
          <svg
            aria-labelledby="Cart"
            class="nut-icon nut-icon-Cart "
            role="presentation"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 65.17c0-24.21 19.37-43.84 43.24-43.84h51.14c70.46 0 128.85 55.4 133.55 126.68l.74 11.12h661.46c81.45 0 144 73.15 132.48 154.9l-33.02 234.45c-14.63 103.94-97.49 184.23-200.58 194.35l-467.47 45.86C246.4 796.07 180.12 739 175.1 662.63l-30.03-456.32v-.76l-3.42-51.69c-1.66-25.24-22.34-44.84-47.27-44.85H43.24C19.35 109.01 0 89.38 0 65.17m234.45 181.66 26.95 409.98c1.77 27.03 25.22 47.21 51.82 44.59l467.47-45.87c63.36-6.19 114.28-55.55 123.29-119.44L937 301.63c4.1-28.93-18.05-54.83-46.87-54.82h-655.7zm64.22 670.5a85.33 85.33 0 1 1-170.67 0 85.33 85.33 0 0 1 170.67 0M896 1002.67A85.33 85.33 0 1 0 896 832a85.33 85.33 0 0 0 0 170.67"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="nut-safe-area nut-safe-area-position-bottom"
    />
  </div>
</div>
`;

exports[`emit click event 1`] = `
<div>
  <div
    class="nut-hoverbutton-container"
  >
    <div
      class="nut-hoverbutton"
    >
      <div
        class="nut-hoverbutton-item-container"
      >
        <div
          class="nut-hoverbutton-item-icon"
        >
          <svg
            aria-labelledby="Cart"
            class="nut-icon nut-icon-Cart "
            role="presentation"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 65.17c0-24.21 19.37-43.84 43.24-43.84h51.14c70.46 0 128.85 55.4 133.55 126.68l.74 11.12h661.46c81.45 0 144 73.15 132.48 154.9l-33.02 234.45c-14.63 103.94-97.49 184.23-200.58 194.35l-467.47 45.86C246.4 796.07 180.12 739 175.1 662.63l-30.03-456.32v-.76l-3.42-51.69c-1.66-25.24-22.34-44.84-47.27-44.85H43.24C19.35 109.01 0 89.38 0 65.17m234.45 181.66 26.95 409.98c1.77 27.03 25.22 47.21 51.82 44.59l467.47-45.87c63.36-6.19 114.28-55.55 123.29-119.44L937 301.63c4.1-28.93-18.05-54.83-46.87-54.82h-655.7zm64.22 670.5a85.33 85.33 0 1 1-170.67 0 85.33 85.33 0 0 1 170.67 0M896 1002.67A85.33 85.33 0 1 0 896 832a85.33 85.33 0 0 0 0 170.67"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="nut-safe-area nut-safe-area-position-bottom"
    />
  </div>
</div>
`;

exports[`prop  1`] = `
<div>
  <div
    class="nut-cell"
    style="border-radius: 6px; align-items: flex-start;"
  >
    <div
      class="nut-cell-left"
    >
      <div
        class="nut-cell-title"
      >
        URL 跳转
      </div>
    </div>
    <div
      class="nut-cell-extra"
    >
      https://m.jd.com/
    </div>
  </div>
</div>
`;

exports[`prop title extra description test 1`] = `
<div>
  <div
    class="nut-cell"
    data-testid="prop"
    style="border-radius: 6px; align-items: flex-start;"
  >
    <div
      class="nut-cell-left"
    >
      <div
        class="nut-cell-title"
      >
        我是标题
      </div>
      <div
        class="nut-cell-description"
      >
        我是描述
      </div>
    </div>
    <div
      class="nut-cell-extra"
    >
      描述文字
    </div>
  </div>
</div>
`;

exports[`slot default test 1`] = `
<div>
  <div
    class="nut-cell"
    style="border-radius: 6px; align-items: flex-start;"
  >
    <div
      class="nut-cell-left"
    >
      <div
        class="nut-cell-title"
      >
        <div>
          自定义内容
        </div>
      </div>
    </div>
    <div
      class="nut-cell-extra"
    >
      描述文字
    </div>
  </div>
</div>
`;

exports[`slot extra 1`] = `
<div>
  <div
    class="nut-cell"
    style="border-radius: 6px; align-items: flex-start;"
  >
    <div
      class="nut-cell-left"
    >
      <div
        class="nut-cell-title"
      >
        Switch
      </div>
    </div>
    <div
      class="nut-cell-extra"
    >
      <div
        class="nut-switch"
      >
        <div
          class="nut-switch-button nut-switch-button-open"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`tabbar height 1`] = `
<div>
  <div
    class="nut-hoverbutton-container"
    style="bottom: 64px;"
  >
    <div
      class="nut-hoverbutton"
    >
      <div
        class="nut-hoverbutton-item-container"
      >
        <div
          class="nut-hoverbutton-item-icon"
        >
          <svg
            aria-labelledby="Cart"
            class="nut-icon nut-icon-Cart "
            role="presentation"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 65.17c0-24.21 19.37-43.84 43.24-43.84h51.14c70.46 0 128.85 55.4 133.55 126.68l.74 11.12h661.46c81.45 0 144 73.15 132.48 154.9l-33.02 234.45c-14.63 103.94-97.49 184.23-200.58 194.35l-467.47 45.86C246.4 796.07 180.12 739 175.1 662.63l-30.03-456.32v-.76l-3.42-51.69c-1.66-25.24-22.34-44.84-47.27-44.85H43.24C19.35 109.01 0 89.38 0 65.17m234.45 181.66 26.95 409.98c1.77 27.03 25.22 47.21 51.82 44.59l467.47-45.87c63.36-6.19 114.28-55.55 123.29-119.44L937 301.63c4.1-28.93-18.05-54.83-46.87-54.82h-655.7zm64.22 670.5a85.33 85.33 0 1 1-170.67 0 85.33 85.33 0 0 1 170.67 0M896 1002.67A85.33 85.33 0 1 0 896 832a85.33 85.33 0 0 0 0 170.67"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="nut-safe-area nut-safe-area-position-bottom"
    />
  </div>
</div>
`;
